lib/chai/core/assertions.js

```diff
@@ -4061,3 +4062,89 @@ Assertion.addProperty('finite', function (msg) {
     'expected #{this} to not be a finite number'
   );
 });
+
+/**
+ * A subset-aware compare function
+ *
+ * @param {unknown} expected
+ * @param {unknown} actual
+ * @returns {void}
+ */
+function compareSubset(expected, actual) {
+  if (expected === actual) {
+    return true;
+  }
+  if (typeof actual !== typeof expected) {
+    return false;
+  }
+  if (typeof expected !== 'object' || expected === null) {
+    return expected === actual;
+  }
+  if (!!expected && !actual) {
```

Comment:
at this point, we already know `expected` is an `object`, so you can avoid the `!!`

```suggestion
  if(!actual) {
    return false;
  }
```

Reply:
Fixed in latest push (force-pushed for rebase to 1 commit).